### DISCUSSION TITLE
How to setup many to many relationship in supabase UI using UUID

### DISCUSSION THREAD

**Discussion by quenginedev** [supabase/supabase#1580] [Questions]
Hello there 👋🏽, I am new to Supabase and Postgres. Whenever I create a many-to-many relationship in Supabase UI, I keep getting the error `Could not find foreign keys between these entities. No relationship found between table1 and table2`. But when I try a one-to-one relationship with works perfectly.

I want to know if I am doing it wrong or I found a bug?

Thanks in advance 😃

**Comment by soedirgo** [comment]
Hey @quenginedev, does this work for you? https://github.com/supabase/supabase/discussions/710

**Reply by quenginedev** [reply]
Thanks for the video @soedirgo, But unfortunately what I'm trying to do is a bit different. I have a table ( Let's call recipes ) with a column of cuisines which is an array of UUIDs,  I would like to be able to get the content of the cuisines array from the cuisines table when I make a query. I tried adding a foreign key to the cuisines column of the recipes table but I keep on getting this error 
`Error updating foreign key: foreign key constraint "recipes_cuisines_fkey" cannot be implemented`.
Please is there a way to make this work?

**Reply by soedirgo** [reply]
So I'm guessing you want `recipes.cuisines` to be an array of UUIDs, each referencing `cuisines.id`? I don't think this is possible (more info [here](https://stackoverflow.com/questions/41054507/postgresql-array-of-elements-that-each-are-a-foreign-key)), the way you do many-to-many relationships in RDMSs is through a third "bridge" table.

Once you set this up though, you can simply query all cuisines for a particular recipe like so (assuming `recipes_cuisines` is the bridge table):
```js
// recipes_cuisines: id | recipe_id | cuisine_id
const { data, error } = await supabase
  .from('recipes_cuisines')
  .select('cuisines(*)')
  .eq('recipe_id', '5ff46b54-b6e8-11eb-a910-c353e6997b8e')
```

**Reply by quenginedev** [reply]
I would create the bridge and let you know if anything comes up. Thanks
